### DISCUSSION TITLE
Running starship on a single thread environment

### DISCUSSION THREAD

**Discussion by pati-ni** [starship/starship#5464] [Q&A]
I want to access a High-Performance-Computing node where multi-threading is not allowed. Therefore, starship is killed upon spawning. Is there a way to disable it from launching several different threads?

Here is how it crashes:

```
thread 'main' panicked at 'Failed to initialize worker thread pool: ThreadPoolBuildError { kind: IOError(Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }) }', src/main.rs:271:10
```

**Comment by chipbuster** [comment]
You can set `STARSHIP_NUM_THREADS` to 1, though I'm not 100% sure if this will work in your environment.
